tests/test_readme.py

```diff
@@ -0,0 +1,18 @@
+import re
+from pathlib import Path
+
+import pytest
+
+
+def _parse_readme_examples():
+    lines = open(Path(__file__).parent.parent / "README.md").read().splitlines()
+
+    # use regex to find all python code blocks
+    code_blocks = re.findall(r"```python(.*?)```", "\n".join(lines), re.DOTALL)
+
+    return code_blocks
+
+
+@pytest.mark.parametrize("example", _parse_readme_examples())
+def test_readme_example(example):
+    exec(example)
```

Comment:
I am not very familiar with the `exec()` function, but I am assuming that `example` is a string? Could we make that explicit in a docstring?

Reply:
good idea, I'll do that

Reply:
done :) could you have another look?